File: internal/model/post.go
package model

import "time"

type Post struct {
	ID        uint      `gorm:"primaryKey"`
	Title     string    `gorm:"type:varchar(255);not null"`
	Content   string    `gorm:"type:text;not null"`
	AuthorID  uint      `gorm:"not null;index"`
	Status    string    `gorm:"type:varchar(20);default:'draft'"`
	CreatedAt time.Time `gorm:"autoCreateTime"`
	UpdatedAt time.Time `gorm:"autoUpdateTime"`
}

func (Post) TableName() string {
	return "posts"
}

File: web/src/types/post.ts
export interface User {
  id: number;
  username: string;
  email: string;
  role: string;
}

export interface Post {
  id: number;
  title: string;
  content: string;
  author_id: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface LoginResponse {
  token: string;
  user: User;
}

export interface PostResponse {
  post: Post;
  message?: string;
}